### ISSUE TITLE
Expected indexing speed?

### ISSUE THREAD

**Issue by taylorthurlow** [Shopify/ruby-lsp#1316]
Trying this out for the first time and for a Rails app with 25-30 dependencies the index is looking to take about an hour. Inspecting `lsof` and the open file handles it only seems to be getting around 3-5 files parsed per second. 

Is there something I'm missing here, or some weird performance bug I might be running into? This is currently running Ruby 3.2 and an M2 Pro Mac Mini. One core is pegged at 100%.

**Comment by taylorthurlow** [comment]
Ok, running the above test and things are done in less than 10 seconds. Also confirmed that `RubyIndexer.configuration.indexables.map(&:full_path)` gives me the list of files that I'd expect it to be indexing, and it does - just short of 10,000 entries.

Can you recommend a next step as far as seeing why running the language server via the executable might behave differently?

**Comment by taylorthurlow** [comment]
> Try temporarily disabling all other Neovim plugins in case something is clashing.

I clearly didn't do a good enough job the first time I tried disabling everything but the relevant neovim LSP built-ins, this is my fault. I ended up narrowing it down to the presence of this line (and something to do with [ray-x/lsp_signature](https://github.com/ray-x/lsp_signature.nvim).

```lua
require("lsp_signature").on_attach({})
```

Sorry to have wasted both of your time on this one.

If anyone else has a similar issue, I was debugging the RPC requests to the language server and this one seemed to trigger it:

```lua
{
    id = 3,
    jsonrpc = "2.0",
    method = "textDocument/signatureHelp",
    params = {
        position = {character = 1, line = 1},
        textDocument = {uri = "file:///Users/taylor/Code/app/config/application.rb"}
    }
}
```

At this point it's not clear to me what is causing the slowdown, but there aren't any extra RPC requests being sent that I can see causing a slowdown.
